presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpillerFactory.java

```diff
@@ -83,9 +88,11 @@ public FileSingleStreamSpillerFactory(
             BlockEncodingSerde blockEncodingSerde,
             SpillerStats spillerStats,
             List<Path> spillPaths,
-            double maxUsedSpaceThreshold)
+            double maxUsedSpaceThreshold,
+            boolean spillCompressionEnabled,
+            boolean spillEncryptionEnabled)
     {
-        this.serdeFactory = new PagesSerdeFactory(requireNonNull(blockEncodingSerde, "blockEncodingSerde is null"), false);
+        this.serdeFactory = new PagesSerdeFactory(requireNonNull(blockEncodingSerde, "blockEncodingSerde is null"), spillCompressionEnabled);
```

Comment:
Would you extract logic to allow enabling/disabling compression for spill into a separate commit?

Reply:
Will do.